src/api/docgen/transpile/python.ts

```diff
@@ -0,0 +1,318 @@
+  }
+
+  public str(): transpile.TranspiledTypeReference {
+    const typing = this.builtins("str");
+    return { raw: typing, markdown: Markdown.pre(typing) };
+  }
+
+  public number(): transpile.TranspiledTypeReference {
+    const types = ["int", "float"];
+    const typing = this.typing("Union");
+    return {
+      raw: `${typing}[${types.join(", ")}]`,
+      markdown: `${Markdown.bold(this.typing("Union"))}[${types
+        .map((t) => Markdown.pre(t))
+        .join(", ")}]`,
+    };
+  }
+
+  public date(): transpile.TranspiledTypeReference {
+    const d = "datetime.datetime";
+    return { raw: d, markdown: Markdown.pre(d) };
+  }
+
+  public enum(enu: reflect.EnumType): transpile.TranspiledEnum {
+    return {
+      fqn: this.type(enu).fqn,
+      name: enu.name,
+    };
+  }
+
+  public enumMember(em: reflect.EnumMember): transpile.TranspiledEnumMember {
+    return {
+      fqn: `${this.enum(em.enumType).fqn}.${em.name}`,
+      name: em.name,
+    };
+  }
+  public json(): transpile.TranspiledTypeReference {
+    const a = "any";
+    return { raw: a, markdown: Markdown.pre(a) };
+  }
+
+  public property(property: reflect.Property): transpile.TranspiledProperty {
+    return {
+      name: property.const ? property.name : toSnakeCase(property.name),
+      parentType: this.type(property.parentType),
+      typeReference: this.typeReference(property.type),
+      optional: property.optional,
+    };
+  }
+
+  public class(klass: reflect.ClassType): transpile.TranspiledClass {
+    return {
+      name: toSnakeCase(klass.name),
+      type: this.type(klass),
+    };
+  }
+
+  public parameter(
+    parameter: reflect.Parameter
+  ): transpile.TranspiledParameter {
+    return {
+      name: toSnakeCase(parameter.name),
+      parentType: this.type(parameter.parentType),
+      typeReference: this.typeReference(parameter.type),
+      optional: parameter.optional,
+    };
+  }
+
+  public struct(struct: reflect.InterfaceType): transpile.TranspiledStruct {
+    const type = this.type(struct);
+    const inputs = struct.allProperties.map((p) =>
+      formatInput(this.property(p))
+    );
+    return {
+      type: type,
+      name: struct.name,
+      import: formatImport(type),
+      initialization: formatInvocation(type, inputs),
+    };
+  }
+
+  public callable(callable: reflect.Callable): transpile.TranspiledCallable {
+    const type = this.type(callable.parentType);
+
+    const parameters = new Array<reflect.Parameter>();
+
+    for (const p of callable.parameters.sort(this.optionalityCompare)) {
+      if (!this.isStruct(p)) {
+        parameters.push(p);
+      } else {
+        // struct parameters are expanded to the individual struct properties
+        const struct = p.parentType.system.findInterface(p.type.fqn!);
+        for (const property of struct.allProperties) {
+          const parameter = propertyToParameter(callable, property);
+          parameters.push(parameter);
+        }
+      }
+    }
```

Comment:
This expands every struct parameter, which is probably wrong. I'm not sure how to identify just the one I need. Can I rely on the order and take the last? @RomainMuller ?

Reply:
I believe that's the behavior of the code generator